File: python/Gpu.py
from Komponen import Komponen
class Gpu(Komponen) :
    def __init__(self, vramGB, architectureType, supportRayTracing, merk, nama):
        super().__init__(merk, nama)
        
        self.__vramGB = vramGB
        self.__architectureType = architectureType
        self.__supportRayTracing = supportRayTracing
        
    def get_vramGB(self):
        return self.__vramGB
    def get_architectureType(self):
        return self.__architectureType
    def get_supportRayTracing(self):
        return self.__supportRayTracing
    
    # Setter methods
    def set_vramGB(self, vramGB):
        self.__vramGB = vramGB
    def set_architectureType(self, architectureType):
        self.__architectureType = architectureType
    def set_suppoerRayTracing(self, suppoerRayTracing):
        self.__suppoerRayTracing = suppoerRayTracing
        

File: cpp/Gpu.cpp
// Gpu.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
using namespace std;

class Gpu : public Komponen {
private:
    int vramGB;
    string architectureType;
    bool supportRayTracing;
    
public:
    Gpu() {}
    
    Gpu(int vramGB, string architectureType, bool supportRayTracing, string merk, string nama) : Komponen(merk, nama) {
        this->vramGB = vramGB;
        this->architectureType = architectureType;
        this->supportRayTracing = supportRayTracing;
    }
    
    void setVramGB(int vramGB) {
        this->vramGB = vramGB;
    }
    
    void setArchitectureType(string architectureType) {
        this->architectureType = architectureType;
    }
    
    void setSupportRayTracing(bool supportRayTracing) {
        this->supportRayTracing = supportRayTracing;
    }
    
    int getVramGB() {
        return this->vramGB;
    }
    
    string getArchitectureType() {
        return this->architectureType;
    }
    
    bool getSupportRayTracing() {
        return this->supportRayTracing;
    }
    
    ~Gpu() {}
};